pkg/types/message_envelope.go

```diff
@@ -27,14 +30,22 @@ const (
 
 // MessageEnvelope is the data structure for messages. It wraps the generic message payload with attributes.
 type MessageEnvelope struct {
```

Comment:
The NATS implementation needs to be update to copy these new fields into/from the NATS header 
here: https://github.com/edgexfoundry/go-mod-messaging/blob/e8d1654ba5965c3fb713dc646ae6e642854fae7d/internal/pkg/nats/marshaller.go#L65-L66
and here: 
https://github.com/edgexfoundry/go-mod-messaging/blob/e8d1654ba5965c3fb713dc646ae6e642854fae7d/internal/pkg/nats/marshaller.go#L77-L78

For `QueryParams` you will need to transform the Go map to/from a list of string values containing the Key/Value pairs and use the `.Header.Add("QueryParams", "<key>:<value>")` to add each one 
and `.Header.Values("QueryParams")` to get the list.